### ISSUE TITLE
Reprojucer.cmake doesn't support the setting PLUGIN_VST3_CATEGORY

### ISSUE THREAD

**Issue by rclement** [McMartin/FRUT#375]
It would be great for `Reprojucer.cmake` to support the setting `PLUGIN_VST3_CATEGORY` ("Plugin VST3 Category" in Projucer).

As of Projucer 5.3.1, this is also a multiple choice value.

Expected output in `AppConfig.h`:
```cpp
#ifndef  JucePlugin_Vst3Category
 #define JucePlugin_Vst3Category           "Fx|Delay"
#endif
```

**Comment by McMartin** [comment]
@rclement 

Thank you very much for opening this issue! I really appreciate that you made the issue title follow the same pattern as #251 and #252.

Making `Reprojucer.cmake` support `PLUGIN_VST3_CATEGORY` is definitely on my radar. However it might require supporting `PLUGIN_FORMATS` and `PLUGIN_CHARACTERISTICS` (also new in Projucer 5.3.1), which are a bit complex to implement, so I can't give any ETA.

**Comment by rclement** [comment]
@McMartin Thanks for your explanation!

I'm sorry but I don't see how `PLUGIN_VST3_CATEGORY` would be related to `PLUGIN_FORMATS` and/or `PLUGIN_CHARACTERISTICS`, at least at FRUT level:

- `PLUGIN_FORMATS` is for the audio plugin format(s) selection: instead of having multiple `BUILD_<FORMAT>` options better to have an extensible list

- `PLUGIN_CHARACTERISTICS` is just some "meta"-options global to all plugin formats (same as before, better to have some extensible list instead of multiple options

-  `PLUGIN_VST3_CATEGORY` is exactly like `PLUGIN_AAX_CATEGORY` : some specific option for a specific audio plugin format

In my opinion, `PLUGIN_VST3_CATEGORY` is just an omission as it should not require some more development than integrating `PLUGIN_AAX_CATEGORY`.

Please let me know if I overlooked some specific issue/behaviour.

**Comment by McMartin** [comment]
@rclement I apologize, `PLUGIN_VST3_CATEGORY` is indeed not related to `PLUGIN_FORMATS` nor `PLUGIN_CHARACTERISTICS`, I got confused by the fact that these Projucer settings were all introduced in the same commit: https://github.com/WeAreROLI/JUCE/commit/629f2a29797a4274c2bd7a8a23e03bb6f814666d.

In that commit, "Plugin VST3 Category" was added to Projucer and the 3 other similar settings ("Plugin VST Category", "Plugin RTAS Category" and "Plugin AAX Category") changed from simple text fields to multi-choice components. I'll need to spend more time thinking about the code to decide whether it's fine to add the simple text field version for VST3 to `Reprojucer.cmake`, or whether I should convert the other 3 settings to multi-choice first.

I'll keep you posted!
